test/HarpoonBattleTest.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * File:   HarpoonBattleTest.cpp
+ * Author: Kjell Hedstrom
+ *
+ * Created on November 30, 2015
+ */
+
+
+#include "HarpoonBattleTest.h"
+#include "Kraken.h"
+#include "KrakenBattle.h"
+#include "HarpoonBattle.h"
+#include "KrakenIntegrationHelper.h"
+#include <Death.h>
+#include <memory>
+#include <tuple>
+#include <utility>
+
+
+TEST_F(HarpoonBattleTest, EnumToString) {
+   auto data = HarpoonBattle::EnumToString(HarpoonBattle::ReceivedType::Data);
+   auto done = HarpoonBattle::EnumToString(HarpoonBattle::ReceivedType::Done);
+   auto error = HarpoonBattle::EnumToString(HarpoonBattle::ReceivedType::Error);
+   auto end = HarpoonBattle::EnumToString(HarpoonBattle::ReceivedType::End);
+
+   EXPECT_TRUE(data == "<DATA>");
+   EXPECT_TRUE(done == "<DONE>");
+   EXPECT_TRUE(error == "<ERROR>");
+   EXPECT_TRUE(end == "<END>");
+
+}
+
+TEST_F(HarpoonBattleTest, EnumToString_Converted) {
+   auto data = HarpoonBattle::EnumToString(static_cast<HarpoonBattle::ReceivedType>(KrakenBattle::SendType::Data));
+   auto done = HarpoonBattle::EnumToString(static_cast<HarpoonBattle::ReceivedType>(KrakenBattle::SendType::Done));
+   auto error = HarpoonBattle::EnumToString(static_cast<HarpoonBattle::ReceivedType>(KrakenBattle::SendType::Error));
+   auto end = HarpoonBattle::EnumToString(static_cast<HarpoonBattle::ReceivedType>(KrakenBattle::SendType::End));
+
+   EXPECT_TRUE(data == "<DATA>");
+   EXPECT_TRUE(done == "<DONE>");
+   EXPECT_TRUE(error == "<ERROR>");
+   EXPECT_TRUE(end == "<END>");
+}
+
+
+// Disabled since it will trigger a fatal shutdown of the logger
+TEST_F(HarpoonBattleTest, EnumToString_Undefined) {
+   Death::SetupExitHandler();
+   Death::ClearExits();
+   EXPECT_FALSE(Death::WasKilled());
+   auto wrong = HarpoonBattle::EnumToString(static_cast<HarpoonBattle::ReceivedType>(1000));
+   auto found = wrong.find("<ERROR>");
+   EXPECT_TRUE(found != std::string::npos) << wrong;
+   EXPECT_TRUE(Death::WasKilled());
+
+   auto foundDeath = Death::Message().find("Unknown ReceivedType:");
+   EXPECT_TRUE(foundDeath != std::string::npos) << Death::Message();
+}
+
+TEST_F(HarpoonBattleTest, StringToEnum) {
+   auto data = HarpoonBattle::StringToEnum("<DATA>");
+   auto done = HarpoonBattle::StringToEnum("<DONE>");
+   auto error = HarpoonBattle::StringToEnum("<ERROR>");
+   auto end = HarpoonBattle::StringToEnum("<END>");
+   auto wrong = HarpoonBattle::StringToEnum("<Hello World>");
+
+   using namespace HarpoonBattle;
+   EXPECT_TRUE(data == ReceivedType::Data);
+   EXPECT_TRUE(done == ReceivedType::Done);
+   EXPECT_TRUE(error == ReceivedType::Error);
+   EXPECT_TRUE(end == ReceivedType::End);
+   EXPECT_TRUE(wrong == ReceivedType::Error);
+}
+
+
+
+TEST_F(HarpoonBattleTest, ChunksExtracted_DATA) {
```

Comment:
Will do .... Although for tests in Research we actually have this a lot.

Personally I like TestName__Expextations but I don't follow that "convention" either here so it will be changed